src/unitxt/utils.py

```diff
@@ -18,6 +20,93 @@ def __call__(cls, *args, **kwargs):
         return cls._instances[cls]
 
 
+class LRUCache:
+    """An LRU (Least Recently Used) cache that stores a limited number of items.
+
+    This cache automatically removes the least recently used item when it
+    exceeds its max size. It behaves similarly to a dictionary, allowing
+    items to be added and accessed using `[]` syntax.
+
+    This implementation is thread-safe, using a lock to ensure that only one
+    thread can modify or access the cache at any time.
+
+    Attributes:
+        max_size (int): The maximum number of items to store in the cache.
+            Items exceeding this limit are automatically removed based on least
+            recent usage.
+    """
+
+    def __init__(self, max_size=10):
+        self._max_size = max_size
+        self._cache = OrderedDict()
+        self._lock = threading.Lock()  # Lock to ensure thread safety
+
+    @property
+    def max_size(self):
+        with self._lock:
+            return self._max_size
+
+    @max_size.setter
+    def max_size(self, size):
+        with self._lock:
+            self._max_size = size
+            # Adjust the cache if the new size is smaller than the current number of items
+            while len(self._cache) > self._max_size:
+                self._cache.popitem(last=False)
+
+    def __setitem__(self, key, value):
```

Comment:
is this needed in addition to `set` below?

Reply:
whats missing there?